mim/commands/search.py

```diff
@@ -561,27 +601,29 @@ def select_by(dataframe: DataFrame,
             highlighted_error(
                 'shown_fields and unshown_fields must be mutually exclusive.'))
 
-    valid_fields = set(dataframe.columns)
+    valid_fields = dataframe.columns
     if shown_fields:
         shown_fields = cast2lowercase(shown_fields)
-        invalid_fields = set(shown_fields) - valid_fields  # type: ignore
+        new_shown_fields, seen_fields = _startswith(valid_fields, shown_fields)
```

Comment:
seen?

Reply:
It is not a good name and I modify the code.